src/lumen/hardware/display.hpp

```diff
@@ -0,0 +1,112 @@
+#pragma once
+
+#include "Adafruit_GFX.h"
+#include "ESP32-HUB75-MatrixPanel-I2S-DMA.h"
+
+#include <memory>
+
+namespace lumen::hardware {
+
+class Display : public Adafruit_GFX {
+public:
+    Display();
+
+    /** Set the brightness of the display.
+     *
+    * \param brightness The brightness value out of. A value of 0 = 0%, and
+    *                   255 = 100%
+    */
+    void setBrightness(uint8_t brightness);
+
+    /** Draw a pixel to a given coordinate.
+     *
+     * \param x X-coordinate of pixel to light.
+     * \param y Y-coordinate of pixel to light.
+     * \param color "Packed" color value of pixel. Use color565() to convert
+     *              typical RGB values into 5:6:5 "packed" format.
+     */
+    void drawPixel(int16_t x, int16_t y, uint16_t color) override;
+
+    /** Draw a pixel to a given coordinate.
+     *
+     * \param x X-coordinate of pixel to light.
+     * \param y Y-coordinate of pixel to light.
+     * \param r Red color value of LED.
+     * \param g Green color value of LED.
+     * \param b Blue color value of LED.
+     */
+    void drawPixelRGB888(int16_t x, int16_t y, uint8_t r, uint8_t g, uint8_t b);
+
+    /** Color every pixel the same color.
+     *
+     * \param color Color value of the screen. Use color444(), color565(), or
+     *              color333() to convert a typical RGB value.
+     */
+    void fillScreen(uint16_t color) override;
+
+    /// Turn off every pixel.
+    void clearScreen();
+
+    /** Convert RGB values into 5:6:5 "packed" 16 bit color format.
+     *
+     * \param red Red color value of LED.
+     * \param green Green color value of LED.
+     * \param blue Blue color value of LED.
+     */
+    static constexpr uint16_t color565(uint8_t red, uint8_t green, uint8_t blue)
```

Comment:
I got linking errors when I separated the implementation from the declaration, but defining the function in the header seemed to resolve them.